cylc/flow/scripts/config.py

```diff
@@ -141,6 +190,8 @@ def main(
         constraint='workflows',
     )
 
+    options_are_valid(options, workflow_id)
+
```

Comment:
Shouldn't this be at the top of the function?

Reply:
Do you mean that the `get_option_parser` should always be directly above the main functions whose wrapper uses it?

I thought that it ought to go at the top, because it acts as a form of internal documentation. `main()` is generally written at the bottom, and that leaves these convenience functions in the middle.

Reply:
The checking of options' validity should be at the beginning of `main()`, otherwise it's missed out. Only problem is that `workflow_id` is not defined until down here too

Reply:
Sorry, I missed your meaning: In general I'd agree, but you've spotted why I can't do that.

Reply:
So what I'm saying is there needs to be a change to allow the validation to come first, otherwise it's not doing its job

Reply:
I've pushed a change to fix this.